core/src/main/java/org/radargun/sysmonitor/AbstractMonitor.java

```diff
@@ -0,0 +1,31 @@
+package org.radargun.sysmonitor;
+
+/**
+ * 
+ * AbstractMonitor is the base class for all system monitors. It manages the lifecycle of the
+ * monitor, so that no values are written to the reports before the benchmark starts or ends.
+ * 
+ */
+public abstract class AbstractMonitor implements Monitor {
+   protected boolean shouldRun = false;
```

Comment:
IIUC the start/stop methods are invoked from main thread, so this should be `volatile`.